### ISSUE TITLE
has_many through association not working

### ISSUE THREAD

**Issue by WWGate** [zdennis/activerecord-import#630]
Trying to import CSV file to the database using "activerecord-import gem". I have the following Models

```
question.rb
require 'csv'
class Question < ApplicationRecord
    has_many :question_answers, dependent: :destroy
    has_many :answers, through: :question_answers
    belongs_to :category
    belongs_to :product
```

```
answer.rb
class Answer < ApplicationRecord
    has_many :question_answers, dependent: :destroy
    has_many :questions, through: :question_answers
 end
```

```
question_answer.rb
class QuestionAnswer < ApplicationRecord
    belongs_to :question
    belongs_to :answer
end
```

The below method is to handle the CSV data and prepare it to be saved using ActiveRecord import gem

```
def self.from_csv(file)
        questions = []
        CSV.foreach(file.path, headers: true) do |row|
            category = Category.find_by(name: row['category'].strip)
            product = Product.find_by(title: row['product'].strip)
            parent_q = Question.find_by(qname: row['parent'])
            question = Question.new(
                question: row['question'],
                qtype: row['qtype'],
                tooltip: row['tooltip'],
                parent: parent_q,
                position: row['position'],
                qname: row['qname'],
                category_id: category.id,
                product_id: product.id,
                state_id: row['state_id'],
                explanation: row['explanation']

            )
            answers = row['answers'].split(" | ") if row['answers'].present?

            if answers.present?
                answers.each do |a_str|
                    answer_arr = a_str.split(',')
                    question.answers.build(answer: answer_arr[0] || "", value: answer_arr[1] || "", pdf_parag: answer_arr[2] || "", key: answer_arr[3] || "", position: answer_arr[4] || "", note: answer_arr[5] || "")
                end
            end
            p question.answers.inspect
            questions << question
        end
        imported_obj = Question.import questions, recursive: true, validate: false
    end
```

the code insert questions but without their answers, it gives an error saying:

`NoMethodError (undefined method `answer_id=' for #<Answer:0x000000000>`
I'm using Heroku

any help is highly appreciated

**Comment by jkowens** [comment]
Unfortunately, this gem does not support importing `has_many :through` as coded above.

I believe something like the following might work:

```
Question.import questions, validate: false

answers = questions.map(&:answers)
Answer.import answers, validate: false

# build an array of question answers objects while iterating the csv file
QuestionAnswer.import question_answers, validate: false
```

**Comment by pilgunboris** [comment]
Hi @jkowens, is it possible to exclude `has_many: through` associations from the list for handling?
We had the same error, and I spent many hours until I found that association reflection returns incorrect `foreign_key` from the proxy model here: https://github.com/zdennis/activerecord-import/blob/82ac6f9c90b216be4f1125b36ba7d0d1608dfa6c/lib/activerecord-import /import.rb#L962
It is quite hard to figure out what is going on without deep debugging.

Or it might make sense to raise an understandable error, so it would be clear why does it fail? 
This might be even better because people might expect these associations to be imported and need to let them know to handle it differently.
